src/cli/serve/serve.js

```diff
@@ -171,10 +173,17 @@ export default function (program) {
     }
 
     process.on('SIGHUP', function reloadConfig() {
-      const settings = getCurrentSettings();
+      const settings = transformDeprecations(getCurrentSettings());
+      const config = new Config(kbnServer.config.getSchema(), settings);
+
       kbnServer.server.log(['info', 'config'], 'Reloading logging configuration due to SIGHUP.');
-      kbnServer.applyLoggingConfiguration(settings);
+      kbnServer.applyLoggingConfiguration(config);
       kbnServer.server.log(['info', 'config'], 'Reloaded logging configuration due to SIGHUP.');
+
+      // If new platform config subscription is active, let's notify it with the updated config.
+      if (kbnServer.newPlatformConfig) {
```

Comment:
There are only two legacy kibana server tests that are failing because of this PR. I haven't fixed them in this PR, but plan to do so in the follow-ups, just need more time to figure out the best way to do so.

1. `src/ui/__tests__/ui_exports_replace_injected_vars.js` - test creates `KibanaServer` with custom plugins path that doesn't include `elasticsearch` and config doesn't include schema for it as well, but it's a requirement for the new platform, fix is easy.

2. `src/cli/serve/__tests__/reload_logging_config.js` - this test is full of magic and fails just because of absence of `tags` property in new platform `LogRecord`, fix is also trivial, but I've deferred that to PR that will introduce `legacy-json` and `legacy-pattern` layouts.

Reply:
:+1: We don't have to do them in this PR.

Reply:
👍